### PR TITLE
feat: add status code support to ErrorWithProps

### LINKED ISSUE #592: Add a statusCode optional parameter to ErrorWithProps
Instead of

```js
const err = new mercurius.ErrorWithProps('UNAUTHORIZED');
err.statusCode = 200
throw err
```

We should be doing

```js
throw new mercurius.ErrorWithProps('UNAUTHORIZED', 200);
```

Note that we should also add some documents on how errorFormatter works as there has been a few issues about it. There might be something more to change under the hood - we might want to return a `200` by default in case of errors for example (sigh).

_Originally posted by @mcollina in https://github.com/mercurius-js/auth/issues/32#issuecomment-932356247_

### PR DESCRIPTION
**Pull request by brainrepo**
closes #592

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -592,11 +592,12 @@ declare namespace mercurius {
    * Extended errors for adding additional information in error responses
    */
   class ErrorWithProps extends Error {
-    constructor(message: string, extensions?: object);
+    constructor(message: string, extensions?: object, statusCode?: number);
```
Comment (jonnydgreen) [inline index.d.ts:595]: could we have a type test for this?

File: lib/errors.js [hunk 2]
```diff
@@ -4,9 +4,10 @@ const { formatError, GraphQLError } = require('graphql')
 const createError = require('fastify-error')
 
 class ErrorWithProps extends Error {
-  constructor (message, extensions) {
+  constructor (message, extensions, statusCode) {
```
Comment (jonnydgreen) [inline lib/errors.js:7]: could we update to the documentation for this?

**Review by jonnydgreen — COMMENTED** [review]
Functionally looks good - just a couple of extra things!

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
@jonnydgreen you good with this?

**Pull request merged**